pkg/cmd/server/bootstrappolicy/policy.go

```diff
@@ -397,6 +397,8 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
 			Rules: []authorizationapi.PolicyRule{
 				// push and pull images
 				authorizationapi.NewRule("get", "update").Groups(imageGroup).Resources("imagestreams/layers").RuleOrDie(),
+				// allow auto-provisioning when pushing an image that doesn't have an imagestream yet
```

Comment:
> does this allow custom builders to create imagestream for arbitrary images?

Yes, but given push rights, they can destroy any existing one already.  The ability to create new ones doesn't do much if nothing is wired to use them.